### PR TITLE
test: optimizer multiflow

### PR DESCRIPTION
**Pull request by florian-hoenicke**

### CODE REVIEW

**Review by JoanFM — COMMENTED** [review]
Could it also be added with `docker` Pods?

File: tests/integration/optimizers/test_optimizer.py [hunk 1]
```diff
@@ -58,14 +58,86 @@ def test_optimizer(tmpdir, config):
...
+            flow_yaml='tests/integration/optimizers/flow1.yml',
+            documents=document_generator(10),
+            request_size=1,
+            execution_method='search',
+        ),
+        SingleFlowRunner(
+            flow_yaml='tests/integration/optimizers/flow2.yml',
+            documents=document_generator(10),
+            request_size=1,
+            execution_method='search',
```
Comment (maximilianwerk) [inline tests/integration/optimizers/test_optimizer.py:73]: I would put the second flow first and remove the evaluator from the second one. This demonstrates, that we only can leverage one evaluator. 

And we could also achieve this by just setting `execution_method='index'` for the first flow. Then we don't even need two flow files.

File: tests/integration/optimizers/test_optimizer.py [hunk 2]
```diff
@@ -58,14 +58,86 @@ def test_optimizer(tmpdir, config):
...
+with:
+  flow_runner: !MultiFlowRunner
+    with:
+      flows: 
+        - !SingleFlowRunner
+          with:
+            flow_yaml: 'tests/integration/optimizers/flow1.yml'
+            documents: {jsonlines_file}
+            request_size: 1
+            execution_method: 'search_lines'
```
Comment (maximilianwerk) [inline tests/integration/optimizers/test_optimizer.py:100]: if this would be `index_lines`, it would already do what I suggest above.

**Review by maximilianwerk — CHANGES_REQUESTED** [review]
I believe we can do the test with one flow file and once call index and once search. This also leverages our index/search setup.

File: tests/integration/optimizers/test_optimizer.py (hunk 1 above)
Comment (florian-hoenicke) [reply tests/integration/optimizers/test_optimizer.py]: nice idea! I changed it accordingly

File: tests/integration/optimizers/test_optimizer.py (hunk 2 above)
Comment (florian-hoenicke) [reply tests/integration/optimizers/test_optimizer.py]: changed, thanks

File: tests/integration/optimizers/test_optimizer.py [hunk 3]
```diff
@@ -60,18 +60,20 @@ def test_optimizer_single_flow(tmpdir, config):
...
-        ),
-        SingleFlowRunner(
-            flow_yaml='tests/integration/optimizers/flow2.yml',
-            documents=document_generator(10),
-            request_size=1,
-            execution_method='search',
-        )
+        [
+            SingleFlowRunner(
+                flow_yaml='tests/integration/optimizers/flow.yml',
```
Comment (JoanFM) [inline tests/integration/optimizers/test_optimizer.py:65]: why not making it independent from where the tests is run. this way is easier to move tests around eventually

Comment (maximilianwerk) [reply tests/integration/optimizers/test_optimizer.py:65]: That is a legacy from me in the optimizer integration tests. Shame on me. I'll fix it.

**Review by maximilianwerk — APPROVED** [review]
LGTM

**Pull request merged**